clippy_lints/src/default.rs

```diff
@@ -103,18 +103,40 @@ impl LateLintPass<'_> for Default {
     }
 
     fn check_block<'tcx>(&mut self, cx: &LateContext<'tcx>, block: &Block<'tcx>) {
-        // find all binding statements like `let mut _ = T::default()` where `T::default()` is the
-        // `default` method of the `Default` trait, and store statement index in current block being
-        // checked and the name of the bound variable
-        let binding_statements_using_default = enumerate_bindings_using_default(cx, block);
-
         // start from the `let mut _ = _::default();` and look at all the following
         // statements, see if they re-assign the fields of the binding
-        for (stmt_idx, binding_name, binding_type, span) in binding_statements_using_default {
-            // the last statement of a block cannot trigger the lint
-            if stmt_idx == block.stmts.len() - 1 {
-                break;
-            }
+        let stmts_head = match block.stmts {
+            [head @ .., _] if !head.is_empty() => head,
```

Comment:
IIUC, this skips the last statement in the block. If so, why? Can you add a comment here, why the last statement is not relevant?

Reply:
I'll add a comment: "Skip the last statement since there cannot possibly be any following statements that re-assign fields". This has the same effect as the if-then-break removed from line 115.